server/src/main/java/org/elasticsearch/search/sort/NestedSortBuilder.java

```diff
@@ -51,6 +53,7 @@ public NestedSortBuilder(StreamInput in) throws IOException {
         path = in.readOptionalString();
         filter = in.readOptionalNamedWriteable(QueryBuilder.class);
         nestedSort = in.readOptionalWriteable(NestedSortBuilder::new);
+        maxChildren = in.readOptionalVInt();
```

Comment:
we need to handle backward compatibility here so you can do:
````
if (in.getVersion.onOrAfter(Version.V_7_0_0_alpha1)) {
   maxChildren = in.readVInt(); // no need to use the optional version
} else {
  maxChildren = Integer.MAX_VALUE;
}
````
and we'll adapt the version check after the backport to 6x